Sources/TuistKit/Services/Inspect/InspecImplicitImportsService.swift

```diff
@@ -83,6 +90,27 @@ final class InspectImplicitImportsService {
         }
     }
 
+    private func explicitTargetDependencies(
+        graphTraverser: GraphTraverser,
+        target: GraphTarget,
+        recursiveSPM: Bool
+    ) -> Set<String> {
+        let targetDependencies = graphTraverser
+            .directTargetDependencies(path: target.project.path, name: target.target.name)
+
+        let explicitTargetDependencies = targetDependencies.map { targetDependency in
+            if targetDependency.graphTarget.project.isExternal, recursiveSPM {
```

Comment:
I think this should be the default behavior – definitely at least for objc dependencies. It's better to be slightly less strict here since transitive implicit imports for ObjC SPM dependencies are fine since we always link the symbols. 

It's not great, but going against the current will cause unnecessary confusion without necessarily helping anyone.

Reply:
Why do you mark objective c as a separate case? The behaviour is the same for swift and objective-c. If this mode was turned on all the time - we wouldn't know about Firebase implicit deps. Now we notified them and at least development team knows about it. If we want to make this behaviour as default one, I suggest we should make flat for strict checking `tuist inspect implicit-dependencies --strict-checking` to detect cases like that. What do you think?

Reply:
>  Now we notified them and at least development team knows about it.

And they [don't plan to fix](https://github.com/firebase/firebase-ios-sdk/issues/13998) it which is fair enough because SPM allows this setup.

Again, I'd be _strongly_ against adding a new flag, moving the complexity to Tuist users. We should find a default that works for the vast majority of cases. Whatever SPM allows, we should allow, too. Anything else is confusing for users.

Reply:
Well, not [not really](https://github.com/firebase/firebase-ios-sdk/pull/14021#issuecomment-2465422154). I just want to provide checking dependencies for those who wants it and thats it. For example, case with[moya](https://github.com/tuist/tuist/pull/6710#issuecomment-2338107913). If recursive checking of dependencies was default choice, [sphanley](https://github.com/sphanley) wouldn't know that he uses Moya implicitly, but in that case its better to include Moya separately explicitly, not as part of RxMoya library. 

After your thoughts about complexity, I changed my mind. I agree with you, that recursive spm checking sholud be by default. What I suggest is to make it by default, but provide opportunity for those who wants to check packages as well. I understand, that final choice is on you, so if you _very strongly_ against flag, I'll remove it:)

Reply:
@fortmarek Do you still want to remove flag?

Reply:
I understand the reasoning, but I would still suggest removing the flag, so we don't encode a hopefully temporary solution in the CLI interface. We can observe what happens at Firebase and making the check strict again in the future.

Reply:
@fortmarek Removed flag

Reply:
Thanks!

Reply:
> If recursive checking of dependencies was default choice, [sphanley](https://github.com/sphanley) wouldn't know that he uses Moya implicitly, but in that case its better to include Moya separately explicitly, not as part of RxMoya library.

For what it's worth, I'll just throw out that I'm still not fully convinced that relying on a transitive import in this case is a wrong or bad thing. For a further example: Realm offers two SPM targets, `Realm` and `RealmSwift`. In their installation instructions, they [suggest the following](https://www.mongodb.com/docs/atlas/device-sdks/sdk/swift/install/#select-the-package-products): 
> * If you use Swift or Swift and Objective-C APIs, add RealmSwift.
> * If you use only Objective-C APIs, add Realm.

In the case where you only add RealmSwift, the Realm target will still be added transitively– so it seems to me that there are at least some vendors who are intentionally using SPM's transitive dependency inclusion to reduce the number of targets you have to explicitly declare. Taking a position that these sorts of "expected" transitive dependencies are an antipattern in all cases may conflict with the guidance being provided by some SPM vendors.

Reply:
> Taking a position that these sorts of "expected" transitive dependencies are an antipattern in all cases may conflict with the guidance being provided by some SPM vendors.

Yeah, that's also my point of view – we should align with SPM's mental model as much as we can, otherwise we will end up confusing developers. Thanks for the feedback 👍

Reply:
One other thing which you may want to consider: in the current inspection implementation, test targets get flagged for implicit dependencies even if they're resolving them from target under test. But in the Tuist docs, the `code sharing` article has [an example which seems to encourage this](https://docs.tuist.dev/en/guides/develop/projects/code-sharing#example): it demonstrates creating a project extension method which generates a framework target and its associated test target, where only the framework target is ever set as a dependency of the test target. It's hard to tell, then, whether the recommendation for "implicitness" is that this should be another valid case, or whether test targets should always declare any external dependencies they're going to import, even if they would be available from the corresponding non-test target.